f/frizzle/comapeo/comapeo_observations.py

```diff
@@ -172,8 +172,38 @@ def download_attachment(url, headers, save_path):
         return None
 
 
-def camel_to_snake(name):
-    return re.sub("([a-z0-9])([A-Z])", r"\1_\2", name).lower()
+def normalize_and_snakecase_keys(dictionary):
+    """
+    Converts the keys of a dictionary from camelCase to snake_case, handling key collisions and truncating long keys.
```

Comment:
Some more test cases for you to add to the tests. I was pleasantly surprised to see these working!

```py
collisions = {
    "abcDef": 1,
    "abc-def": 2,
    "AbcDef": 3
}

newdict = normalize_and_snakecase_keys(collisions)
assert newdict == {'abc_def': 1, 'abc_def_2': 2, 'abc_def_3': 3}
```